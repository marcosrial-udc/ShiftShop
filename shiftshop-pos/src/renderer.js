import * as React from 'react';
import * as ReactDOM from 'react-dom';
import {Provider} from 'react-redux';
import {MuiThemeProvider} from '@material-ui/core';
import {SnackbarProvider} from 'notistack';
import {IntlProvider} from 'react-intl';

/* Async/await regenerator necesaries */
import 'core-js/stable';
import 'regenerator-runtime/runtime';

import './assets/scss/index.scss';
import theme from './theme'

import {App} from './modules/app';

import configureStore from './store';
import {initReactIntl} from "./i18n";
import './polyfills';

/* Configure store. */
const store = configureStore();

/* Configure i18n. */
const {locale, messages} = initReactIntl();

ReactDOM.render(
    <MuiThemeProvider theme={theme}>
      <Provider store={store}>
        <IntlProvider locale={locale} messages={messages}>
          <SnackbarProvider anchorOrigin={{vertical: 'top', horizontal: 'center'}}
                            autoHideDuration={2500}>
            <App/>
          </SnackbarProvider>
        </IntlProvider>
      </Provider>
    </MuiThemeProvider>,
    document.getElementById("root"));
