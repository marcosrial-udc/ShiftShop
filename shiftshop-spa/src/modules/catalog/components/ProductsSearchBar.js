import React from 'react';
import {FormattedMessage} from 'react-intl';
import PropTypes from 'prop-types';
import {Button, Hidden, TextField} from '@material-ui/core';
import {Search} from '@material-ui/icons';

import useStyles from '../styles/ProductsSearchBar';

import CategorySelector from './CategorySelector';

const ProductsSearchBar = ({keywords, category, handleChangeKeywords, handleChangeCategory, searching, onSearch}) => {
    const classes = useStyles();

    const handleSubmit = event => {
        event.preventDefault();
        onSearch();
    };

    return (
        <form className={classes.form} onSubmit={handleSubmit}>
            <CategorySelector className={classes.categorySelect} allCategories
                              selectedCategory={category} handleSelectedCategory={handleChangeCategory}
                              variant="outlined" margin="dense" fullWidth/>
            <TextField className={classes.searchText} label={<FormattedMessage id="project.global.field.keywords"/>}
                       value={keywords} onChange={handleChangeKeywords}
                       variant="outlined" margin="dense" fullWidth/>
            <Button className={classes.searchButton} size="medium" variant="contained" color="primary"
                    type="submit" disabled={searching}>
                <Search fontSize="small"/><Hidden smUp>&nbsp;<FormattedMessage id="project.global.button.search"/></Hidden>
            </Button>
        </form>
    );
};

ProductsSearchBar.propTypes = {
    keywords: PropTypes.string.isRequired,
    category: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    handleChangeKeywords: PropTypes.func.isRequired,
    handleChangeCategory: PropTypes.func.isRequired,
    searching: PropTypes.bool.isRequired,
    onSearch: PropTypes.func.isRequired
};


export default ProductsSearchBar;
