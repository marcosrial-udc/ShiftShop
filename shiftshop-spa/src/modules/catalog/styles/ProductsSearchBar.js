import {makeStyles} from '@material-ui/core';

const useStyles = makeStyles(theme => ({
    form: {
        [theme.breakpoints.up('sm')]:{
            display: 'flex'
        },
    },
    categorySelect: {
        flex: '1',
        [theme.breakpoints.up('sm')]: {
            '& fieldset': {
                borderTopRightRadius: 0,
                borderBottomRightRadius: 0,
            },
        },
    },
    searchText: {
        flex: '2',
        [theme.breakpoints.up('sm')]: {
            '& fieldset': {
                borderRadius: 0,
            },
        },
    },
    searchButton: {
        flex: '0',
        height: theme.spacing(5),
        marginTop: theme.spacing(1),
        marginBottom: theme.spacing(0),
        boxShadow: 'none !important',
        [theme.breakpoints.down('sm')]: {
            width: '100%'
        },
        [theme.breakpoints.up('sm')]: {
            borderTopLeftRadius: 0,
            borderBottomLeftRadius: 0,
        },
    },
}));

export default useStyles;
