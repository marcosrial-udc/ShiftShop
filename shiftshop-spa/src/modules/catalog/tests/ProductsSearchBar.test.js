import React from 'react';
import renderer from 'react-test-renderer';
import {IntlProvider} from 'react-intl';

import {initReactIntl} from "../../../i18n";

import ProductsSearchBar from '../components/ProductsSearchBar';

jest.mock('../components/CategorySelector', () => () => (<mockCategorySelector/>));

const {messages} = initReactIntl();

describe('ProductsSearchBar snapshot', () => {

    const content = {keywords: 'test', category: 1};

    const enabledSearching = {...content, searching: false};
    const disabledSearching = {...content, searching: true};

    const createProductsSearchBar = ({keywords, category, searching}) => {

        return renderer.create(
            <IntlProvider locale='en' messages={messages}>
                <ProductsSearchBar keywords={keywords} category={category} searching={searching}
                                   handleChangeKeywords={jest.fn()} handleChangeCategory={jest.fn()}
                                   onSearch={jest.fn()}/>
            </IntlProvider>
        );

    };

    test('With values and enabled search', () => {

        const productsSearchBar = createProductsSearchBar(enabledSearching);

        expect(productsSearchBar.toJSON()).toMatchSnapshot();

    });

    test('Disabled search', () => {

        const productsSearchBar = createProductsSearchBar(disabledSearching);

        expect(productsSearchBar.toJSON()).toMatchSnapshot();

    });

});
