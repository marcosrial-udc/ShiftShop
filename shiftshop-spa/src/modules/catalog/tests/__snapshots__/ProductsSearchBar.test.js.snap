// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsSearchBar snapshot Disabled search 1`] = `
<form
  className="makeStyles-form-1"
  onSubmit={[Function]}
>
  <mockCategorySelector />
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-searchText-3 MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
      data-shrink={false}
    >
      Keywords
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-63 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-65"
        >
          <span>
            Keywords
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-searchButton-4 MuiButton-containedPrimary Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </span>
  </button>
</form>
`;

exports[`ProductsSearchBar snapshot With values and enabled search 1`] = `
<form
  className="makeStyles-form-1"
  onSubmit={[Function]}
>
  <mockCategorySelector />
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-searchText-3 MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
      data-shrink={false}
    >
      Keywords
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-63 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-65"
        >
          <span>
            Keywords
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-searchButton-4 MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</form>
`;
